### DISCUSSION TITLE
Styling table

### DISCUSSION THREAD

**Discussion by McfarlaneRob** [zauberzeug/nicegui#885] [Q&A] [answered]
### Question

I've created an app which displays a Table on a page, with some forms for inputting data which then populates the initially empty table.
But am now getting stuck with how to style the rows of the table.
I have managed some general styling by using the .style and .classes of the table (background colour, width, margins), but can't get this to affect the font size or weight within the table, and am not sure how I can style each row in the table. (I'm looking to add a gradient background ideally on each row.

If anyone can point me in the right direction or let me know what I'm missing it would be much appreciated.

The rough structure of the page is:
```python
@ui.page("/")
def page(client: Client):

client.content.classes(add="q-pa-none")

ui.add_head_html('''
                        <style>
                            body {
                                background-color: #1D1D1B;
                            }
                        </style>
                    ''')

TABLE = ui.table(columns=COLUMNS, rows=RESULTS, row_key="id", selection="multiple", pagination=25) \
        .style("margin: auto; width: 100%; background-color: #1D1D1B; font-weight: bold; font-size: 3em") \
        .classes("text-xl text-pink")

ui.run()
```
but only some of the style and classes passed are applying properly

**Comment by falkoschindler** [comment, marked as answer]
A rather simple approach would be to give the table some custom class and use it to style it with CSS defined via `add_head_html`:

```py
ui.add_head_html('''
    <style>
        .my-table {
            border: 1px solid black;
        }
        .my-table tr {
            background: linear-gradient(0deg, white, steelblue);
        }
    </style>
''')

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol'},
]
ui.table(columns=columns, rows=rows, row_key='name').classes('my-table')
```
